shotover-proxy/tests/cassandra_int_tests/mod.rs

```diff
@@ -0,0 +1,72 @@
+use cassandra_cpp::*;
+use std::thread;
+use std::thread::sleep;
+use std::time::Duration;
+use tracing::info;
+
+mod basic_driver_tests;
+
+pub struct CassandraTestContext {
+    pub session: Session,
+}
+
+impl CassandraTestContext {
+    pub fn new() -> Self {
+        CassandraTestContext::new_with_points("127.0.0.1")
+    }
+
+    pub fn new_with_points(contact_points: &str) -> Self {
+        CassandraTestContext::new_with_points_and_port(contact_points, 9042)
+    }
+
+    pub fn new_with_points_and_port(contact_points: &str, port: u16) -> Self {
+        let mut cluster = Cluster::default();
+        cluster.set_contact_points(contact_points).unwrap();
+        cluster.set_port(port).ok();
+        cluster.set_load_balance_round_robin();
+        let mut session;
+        //let query = stmt!("SELECT keyspace_name FROM system_schema.keyspaces;");
+        let query = stmt!("SELECT release_version FROM system.local");
+
+        let attempts = 30;
+        let mut current_attempt = 0;
+
+        loop {
```

Comment:
Now that we've sorted out all the cluster init problems I ran the test without the retry logic here and the test still passes.
So seems we can remove it.

Reply:
oh right, redis connections do have a system to handle shotover port not being ready yet.
This equivalent of a retry used by redis has a few advantages:
* its simpler
* its reusable across different protocols
* it only fails if the port is closed. This is important because if shotover can be in a state where the port is open but the connection fails then the test should fail!

So we should be using this functionality instead of a retry.

Using it just requires the following change + fixing other uses of wait_for_socket_to_open
![image](https://user-images.githubusercontent.com/5120858/141390633-55a2fb7a-3d5c-4e61-a711-6477725f0956.png)

Reply:
this is a test retry mechanism cleanup - not something I want to block on given this PR has taken a bit to get through let's do this in another PR, but worth the cleanup

@rukai given you've already written the code can you just PR once this is merged.

Reply:
sure, lets do that